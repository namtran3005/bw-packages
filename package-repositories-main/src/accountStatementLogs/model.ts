import mongooseLeanId from 'mongoose-lean-id';
import { Document, Schema } from "mongoose";
import { mainConnection } from '../mainConnection';

export interface AccountStatementLogsDoc extends Document {
  id: string;
  userId: string;
  locale: string;
  firstName: string;
  solarisPersonSolarisId: string;
  solarisAccountSolarisId: string;
  email: string;
  sent: boolean;
  reason: string;
  createdAt: Date;
  updatedAt: Date;
}

export const accountStatementLogsSchema: Schema = new Schema(
  {
    userId: {
      type: String,
      required: true,
      index: true,
      unique: true
    },
    locale: {
      type: String,
    },
    firstName: {
      type: String,
    },
    solarisPersonSolarisId: {
      type: String,
      index: true,
    },
    solarisAccountSolarisId: {
      type: String,
      index: true,
    },
    email: {
      type: String,
    },
    sent: {
      type: Boolean,
    },
    reason: {
      type: String,
    }
  },
  { 
    timestamps: true,
    collection: "account-statement-logs",
  }
);

accountStatementLogsSchema.plugin(mongooseLeanId);

export const AccountStatementLogsModel = mainConnection.db.model<AccountStatementLogsDoc>(
  'AccountStatementLogs',
  accountStatementLogsSchema
);
