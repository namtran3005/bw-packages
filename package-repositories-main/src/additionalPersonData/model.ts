import mongooseLeanId from 'mongoose-lean-id';
import {
  additionalPersonDataSchema,
  AdditionalPersonDataDoc,
} from '@bitwala-cryptobank-squad/package-schemas';

import { mainConnection } from '../mainConnection';

additionalPersonDataSchema.plugin(mongooseLeanId);

export const AdditionalPersonDataModel = mainConnection.db.model<AdditionalPersonDataDoc>(
  'AdditionalDataPerson',
  additionalPersonDataSchema
);
