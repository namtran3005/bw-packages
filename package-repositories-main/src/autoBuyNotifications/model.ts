import mongooseLeanId from 'mongoose-lean-id';
import {
  autoBuyNotificationsSchema,
  AutoBuyNotificationsDoc,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

autoBuyNotificationsSchema.plugin(mongooseLeanId);

export const AutoBuyNotificationsModel = mainConnection.db.model<
  AutoBuyNotificationsDoc
>('AutoBuyNotifications', autoBuyNotificationsSchema);
