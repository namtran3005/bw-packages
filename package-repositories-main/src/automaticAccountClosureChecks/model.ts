import {
  AutomaticAccountClosureChecksDoc,
  automaticAccountClosureChecksSchema,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

export const AutomaticAccountClosureChecksModel = mainConnection.db.model<
AutomaticAccountClosureChecksDoc
>('AutomaticAccountClosureChecks', automaticAccountClosureChecksSchema);
