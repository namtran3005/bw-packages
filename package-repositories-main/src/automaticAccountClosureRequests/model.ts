import {
  AutomaticAccountClosureRequestsDoc,
  automaticAccountClosureRequestsSchema,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

export const AutomaticAccountClosureRequestsModel = mainConnection.db.model<
AutomaticAccountClosureRequestsDoc
>('AutomaticAccountClosureRequests', automaticAccountClosureRequestsSchema);
