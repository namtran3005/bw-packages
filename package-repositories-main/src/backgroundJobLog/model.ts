import mongooseLeanId from 'mongoose-lean-id';
import {
  backgroundJobLogsSchema,
  BackgroundJobLogDoc,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

backgroundJobLogsSchema.plugin(mongooseLeanId);

export const BackgroundJobLogModel = mainConnection.db.model<BackgroundJobLogDoc>(
  'BackgroundJobLog',
  backgroundJobLogsSchema
);
