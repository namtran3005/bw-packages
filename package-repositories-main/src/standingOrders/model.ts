import mongooseLeanId from 'mongoose-lean-id';
import {
  StandingOrderDoc,
  standingOrderSchema,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

standingOrderSchema.plugin(mongooseLeanId);

export const StandingOrdersModel = mainConnection.db.model<StandingOrderDoc>(
  'StandingOrders',
  standingOrderSchema
);
