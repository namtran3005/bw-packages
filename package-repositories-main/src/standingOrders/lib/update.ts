import { StandingOrder } from '@bitwala-cryptobank-squad/package-solaris';
import { DocumentDefinition } from 'mongoose';
import { StandingOrdersModel } from '../model';

export const update = async (
  owner: string,
  canceledSO: StandingOrder & { owner: string }
): Promise<DocumentDefinition<void> | DocumentDefinition<void>[] | null> => {
  return StandingOrdersModel.update(
    { owner, solarisId: canceledSO.solarisId },
    {
      $set: canceledSO,
    }
  )
    .lean()
    .exec();
};
