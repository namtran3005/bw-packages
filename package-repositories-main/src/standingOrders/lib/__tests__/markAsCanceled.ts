/* eslint-disable @typescript-eslint/no-explicit-any */
import { StandingOrdersModel } from '../../model';

import { update } from '../update';

describe('insertStandingOrder', () => {
  const exec = jest.fn();
  const lean = jest.fn(() => ({ exec }));

  beforeAll(() => {
    jest
      .spyOn(StandingOrdersModel, 'update')
      .mockImplementation(() => ({ lean } as any));
  });
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should use create method and return the promise', async () => {
    await update('SatoshiNakamotoWasHere', {
      solarisId: 'WarrenBuffet',
    } as any);
    expect(StandingOrdersModel.update).toMatchSnapshot();
  });
});
