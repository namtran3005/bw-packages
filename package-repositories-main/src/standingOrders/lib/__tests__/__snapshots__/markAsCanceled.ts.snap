// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insertStandingOrder should use create method and return the promise 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "owner": "SatoshiNakamotoWasHere",
        "solarisId": "WarrenBuffet",
      },
      Object {
        "$set": Object {
          "solarisId": "WarrenBuffet",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "lean": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Object {
                "exec": [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
            },
          ],
        },
      },
    },
  ],
}
`;
